### ISSUE TITLE
Should ren::String be willing to implicitly coerce into a std::string?

### ISSUE THREAD

**Issue by hostilefork** [metaeducation/ren-cpp#6]
Currently this works:

```
void someFunction(std::string str) {
    ...
}

ren::String renstr {"Hello"};
someFunction(renstr);
```

Should it?  It feels a little weird to have a `.toString()` method on it.  It could also be required that you explicitly cast.

The only strange thing about this vs. the implicit coercions to int and float etc, is that it has no analogue to the constructions in #2.  So it's not possible to get a ren::String implicitly from a C string or std::string... thus the implicit conversion on the reverse feels asymmetrical.

**Comment by hostilefork** [comment]
The biggest problem with this to date so far has been the difficulty in realizing when a std::String or QString isn't going to be treated as a ren::String in block loading, due to the implicit constructors on Loadable.

```
std::string str {"Some [Data] I want to parse..."};

 // some # of lines later...

runtime("either parse", str, rules, "[print {success}] [print {failure}]");
```

That is turning out to cause a lot of errors, while the literal runs are not creating the same problem being "right there" with the code.

However, people will want to be able to have loadable code in strings.  That should be at the callsite though.  There's a technical problem with this using Loadable, but I think this is more in the right spirit:

```
 std::string branches {"[print {success}] [print {failure}]"};
 runtime("either parse", str, rules, Loadable {branches});
```

It suggests something I feared, namely that Loadable needs to be exposed to the user.  Here it doesn't work if the underlying class for the input is Loadable too...because it couldn't explicitly and implicitly construct differently for std::string.

So things have to change around a little bit, but I now feel I like str is a string class it acts (by default) like a string value.  Because that's the usual model, and it seems like you should have to make a conscious effort to treat it another way.

**Comment by hostilefork** [comment]
I think this is resolved, and didn't even have to expose Loadable.  The trick is that if you want your string class to be LOADed instead of treated as a ren::String, then you just extract a const char \* of UTF8 data out of it.  It's a little awkward to do that with a QString because they don't _have_ a const char \* available to give you... you make a QByteArray, ask for UTF8, and then use the byte array's data().  But Qt programmers are used to having to do that, and std::string just has a .cstr() method.

Long story short: it's a good enough solution to let const char \* be the Loadable.
